rancher2/resource_rancher2_cluster.go

```diff
@@ -733,19 +744,12 @@ func getClusterKubeconfig(c *Config, id, origconfig string) (*managementClient.G
 				Actions: cluster.Actions,
 			}
 			err = client.APIBaseClient.Action(managementClient.ClusterType, action, clusterResource, nil, kubeConfig)
-			if err == nil {
-				if isRancher26 && len(token) > 0 {
-					newConfig, err := replaceKubeConfigToken(c, kubeConfig.Config, token)
-					if err != nil {
-						return nil, err
-					}
-					kubeConfig.Config = newConfig
+			if err != nil {
+				if !IsNotFound(err) && !IsForbidden(err) && !IsServiceUnavailableError(err) {
+					return nil, fmt.Errorf("Getting cluster Kubeconfig: %v", err)
```

Comment:
Minor: I'd just `%w` here just so the caller can check the error too if they want.
```suggestion
					return nil, fmt.Errorf("Getting cluster Kubeconfig: %w", err)
```